command/agent/agent.go

```diff
@@ -59,6 +59,8 @@ type Agent struct {
 	httpLogger log.Logger
 	logOutput  io.Writer
 
+	EnterpriseAgent *EnterpriseAgent
```

Comment:
Some comments for Agent and EnterpriseClient would be very helpful.  Given that there are enterprise features that live in OSS, readers will miss the contexts for them.